pkg/cloudprovider/cloudapi/azure/azure_nsg_rules.go

```diff
@@ -162,7 +162,10 @@ func convertIngressToNsgSecurityRules(appliedToGroupID *securitygroup.CloudResou
 
 		srcApplicationSecurityGroups, err := convertToAzureApplicationSecurityGroups(rule.FromSecurityGroups, agAsgMapByNepheControllerName)
 		if err != nil {
-			return []*armnetwork.SecurityRule{}, err
+			if !isRemove {
```

Comment:
lets revisit this. dont like the idea of passing flag in convert function. Allowing this checkin for now